src/screens/SignIn/index.tsx

```diff
@@ -1,13 +1,65 @@
-import React, { FC } from 'react';
-import fire from '../../services/firebase';
-
-const SignIn: FC = () => (
-  <div
-    onClick={() => {
-      console.log(fire);
-    }}>
-    Sign In
-  </div>
-);
+import React, { FC, useEffect } from 'react';
+import { fireApi } from '../../services/firebase';
+import { routes } from '../../router';
+import { Link } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { userActions, userOperations, userSelectors } from '../../redux/user';
+import { RootState } from '../../redux/rootReducer';
+import { Redirect } from 'react-router-dom';
+import {
+  FacebookLoginButton,
+  GoogleLoginButton,
+} from 'react-social-login-buttons';
+import EmailAndPasswordSignIn from './EmailAndPasswordSignIn';
+
+const SignIn: FC = () => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(userOperations.checkSocialMediaRedirect());
+  }, [dispatch]);
+
+  const googleSignIn = () => {
+    dispatch(userActions.loginRequest());
+    fireApi.doGoogleSignIn(fireApi.googleProvider);
+  };
+
+  const facebookSignIn = () => {
+    dispatch(userActions.loginRequest());
+    fireApi.doFacebookSignIn(fireApi.facebookProvider);
+  };
+
+  const userId = useSelector((state: RootState) =>
+    userSelectors.getUserId(state)
+  );
+  const isLoading = useSelector((state: RootState) =>
+    userSelectors.getUserIsLoading(state)
+  );
+  const error = useSelector((state: RootState) =>
+    userSelectors.getUserError(state)
+  );
+
+  return (
+    <>
+      {userId ? <Redirect to={routes.homePagePath} /> : null}
+
+      {isLoading ? (
+        'loading'
+      ) : (
+        <>
+          <EmailAndPasswordSignIn />
+          {error ? error : null}
+          <GoogleLoginButton onClick={googleSignIn} />
+          <FacebookLoginButton onClick={facebookSignIn} />
```

Comment:
<EmailAndPasswordSignIn />
{error ? error : null}
          <GoogleLoginButton onClick={googleSignIn} />
          <FacebookLoginButton onClick={facebookSignIn} />

When it comes to the login error, I think I will do something like: if error and you clicked on login with email and psw, show error under email and psw form. if error and you clicked on fb login, show error under fb and so on. I will keep the "clicked on" logic on the local component's state. anytime I move navigate some other page or there is a successful login, the clicked on logic will be set to false